packages/venia-ui/lib/components/Header/cartTrigger.js

```diff
@@ -40,10 +41,10 @@ const CartTrigger = props => {
         <span className={classes.counter}>{itemCountDisplay}</span>
     ) : null;
 
-    // Because this button behaves differently on desktop and mobile
-    // we render two buttons that differ only in their click handler
-    // and control which one displays via CSS.
-    return (
+    const cartTrigger = hideCartTrigger ? null : (
```

Comment:
Another way to hide the cart trigger is to implement this conditional in the `Header` component instead. But, the if this conditional is placed in the `Header`, the `CartTrigger` would never render in the `checkout` page and the GQL call in the `CartTrigger` will not be fired, due to which the `checkout` page will show an everlasting loading icon. The checkout page relies on the result of the `CartTrigger`'s GQL call to know whether to render an empty state or the `Checkout` component.

The other solution would be to move/add the GQL in the checkout page as well and then conditionally render `CartTrigger` in the `Header` component.